config/config_helpers.go

```diff
@@ -16,12 +17,13 @@ import (
 	"github.com/gruntwork-io/terragrunt/util"
 )
 
-var INTERPOLATION_PARAMETERS = `(\s*"[^"]*?"\s*,?\s*)*`
-var INTERPOLATION_SYNTAX_REGEX = regexp.MustCompile(fmt.Sprintf(`\$\{\s*\w+\(%s\)\s*\}`, INTERPOLATION_PARAMETERS))
+var INTERPOLATION_QUOTED_STRING = `"(?:\\.|[^\\"])*?"`
+var INTERPOLATION_QUOTED_REGEX = regexp.MustCompile(INTERPOLATION_QUOTED_STRING)
+var INTERPOLATION_PARAMETERS = fmt.Sprintf(`\s*(?:%s(?:\s*,\s*%s)*)?\s*`, INTERPOLATION_QUOTED_STRING, INTERPOLATION_QUOTED_STRING)
+var INTERPOLATION_PARAMETERS_REGEX = regexp.MustCompile(fmt.Sprintf(`^%s$`, INTERPOLATION_PARAMETERS))
+var INTERPOLATION_SYNTAX_REGEX = regexp.MustCompile(fmt.Sprintf(`\$\{\s*(\w+)\((%s)\)\s*\}`, INTERPOLATION_PARAMETERS))
```

Comment:
Finally, this adds the rest of the "stuff" around a properly formatted interpolation call.

The general pattern is: `\$\{(\w+)\(A)\)\}`, which should match `${anything(parameter-list)}`.  Again, throw in a bunch of `\s*`s and we have our pattern.

![image](https://user-images.githubusercontent.com/766820/59022185-e25c4f00-8812-11e9-8e1b-b4bea8ff9b92.png)

Note, I capture the \w+ and the contents of the parameter list.

Reply:
With the regex's set up, note that we now have a regex that can accurately describe any properly formatted interpolation string in the format of `${foo_bar("a", "b", "c")}`.   HCL/HIL actually allows nesting, which I don't think that this will properly handle, but that's where we would have to go to a real parser as opposed to regex hackery.

Now, I no longer need to create additional regexes to help me validate or extract my strings.  I should be able to:

1. Detect an interpolation string with `INTERPOLATION_SYNTAX_REGEX`
2. Extract match[1] as the function name
3. Extract match[2] as the full parameter list
4. Use `INTERPOLATION_QUOTED_REGEX` to extract all my parameters.

This is exactly what the rest of the code attempts to do.